Tabnine/src/main/java/com/tabnine/chat/ChatBundleExtractor.kt

```diff
@@ -0,0 +1,49 @@
+package com.tabnine.chat
+
+import com.intellij.openapi.diagnostic.Logger
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
+import java.io.BufferedInputStream
+import java.io.File
+import java.io.InputStream
+import java.nio.file.Path
+import kotlin.jvm.Throws
+
+private const val CHAT_BUNDLE_TAR_FILE = "chat-bundle.tar.gz"
+
+object ChatBundleExtractor {
+
+    @Throws(IllegalStateException::class)
+    fun extractBundle(destination: Path) {
+        val resource = javaClass.classLoader.getResourceAsStream(CHAT_BUNDLE_TAR_FILE)
+            ?: throw IllegalStateException("Unable to find resource $CHAT_BUNDLE_TAR_FILE")
+        untar(resource, destination.toFile())
```

Comment:
why is it extracted at runtime and not  in the CI as we did in vscode? @yairco1990

Reply:
because `javaClass.classLoader.` doesn't have a f.s.-like API, you can't "get all resources under x", you need to find each resource by name.

And since these files include js map files and such, I decided that the best solution would be to bundle a single resource which is the tar, and simply untar it at runtime